README.md

```diff
@@ -15,21 +19,53 @@ The artefact mainly consists of a library written in the Haskell language that i
 The library is accompanied with a demo application that uses the library to reproduce our motivating example (Alice and Bob using feature trace recording in Section 2 in our paper) as well as examples of the edit patterns we used to evaluate feature trace recording (Section 5).
 
 
+## What is Feature Trace Recording?
+
+Feature trace recording is a semi-automated method for recording feature-to-code mappings during software development.
+It unlocks evolution as a new source of information for feature location.
+
+![Feature Trace Recording Example](meta/FeatureTraceRecording.png)
+
+Consider a developer Alice editing the `pop` method of a `Stack` class in Java as depicted in the image above.
+Feature-to-code mappings are indicated by corresponding colours.
+By labeling her edits with the feature she edits, we derive feature mappings for edited source code.
+We refer to such a label as [_feature context_](https://pmbittner.github.io/FeatureTraceRecording/docs/FeatureTraceRecording.html#t:FeatureContext).
+It can be left empty when developers do not know the feature they edit.
+In our example, Alice does not know the feature of the moved line in her second edit and omits the feature context (i.e., sets it to null).
+
+From feature contexts we can compute feature mappings for the current code base but also for other variants.
+Assume there is a second developer, Bob, working on another variant (e.g., branch or fork) of the software.
+Bob’s variant might implement other features than Alice’s.
+Assume Bob’s variant implements the blue feature <span style="color:#00BEEB">SafeStack</span> but not the orange feature <span style="color:#D95F02">ImmutableStack</span>.
+Although Alice made many edits to <span style="color:#D95F02">ImmutableStack</span>, we can still derive feature mappings for Bob’s code.
+When Alice deletes code from <span style="color:#D95F02">ImmutableStack</span>, we can infer that this code does not belong to <span style="color:#D95F02">ImmutableStack</span> anymore.
+We thus introduce the new feature mapping <span style="color:#7570B3">¬ImmutableStack</span> for the deleted code that is still present in Bob’s variant (highlighted in purple):
+
+<img src="meta/BobsVariant.png" alt="Bob's Recorded Feature Traces" width="250" />
+
+A detailed explanation of this example can be found in our [paper][paper] and [preprint][preprint].
```

Comment:
The link for `paper` is leading to a DOI not found page, but the DOI of the paper has just not been enabled yet, right?

Reply:
yes. This is the DOI ACM gave us. The paper should be available under this DOI once its published.